src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +73,13 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> workablePredicate = p -> p.getAge() >= fromAge
+                && ((p.getSex() == Person.Sex.MAN && p.getAge() <= maleToAge)
+                || (p.getSex() == Person.Sex.WOMAN && p.getAge() <= femaleToAge));
+
+        return peopleList.stream()
+                .filter(workablePredicate)
+                .collect(Collectors.toList());
```

Comment:
This is overcomplicated, you can do it like this for example:

```suggestion
        return peopleList.stream()
                .filter(p -> p.getAge() >= fromAge
                        && (p.getSex() == Person.Sex.MAN
                        ? p.getAge() <= maleToAge
                        : p.getAge() <= femaleToAge))
                .toList();```